### DISCUSSION TITLE
Some Ploblems abount License and trademark

### DISCUSSION THREAD

**Discussion by snowdream** [drakkan/sftpgo#1836] [Q&A] [answered]
I want to package sftpgo into docker images.
https://github.com/snowdreamtech/sftpgo

1. can my project named with sftpgo?
2. which License my project should use?  AGPL V3? GPL V3? or other license?
3. can i use sftpgo for Commercial use?
4. can i use sftpgo in Self-Host Server? For Personal Use, Or SAAS?
5. If i use a docker panel (eg. Portainer) to pull and run a docker image which contains sftpgo, then Portainer should licensed with AGPL V3?

**Comment by drakkan** [comment]
Please read our [compliance](https://sftpgo.com/compliance.html) page.

The only unanswered question is about Portainer, but of course the SFTPGo license does not change the Portainer license.

You asked for many things, from personal use to reselling SFTPGo as SaaS (which you obviously can't do), so what I'll write below is not necessarily aimed at you.

Leaving aside the legal standpoint, let's consider it from a social point of view. Do you think that as the author of SFTPGo, a project I've been working on since 2019 many hours a day, I could be happy to see companies basing their business on my software free riding on my generosity? How would you feel in my place? What will you do if one of your users asks for support and you are unable to provide it? You will likely open an issue on GitHub tryng to free riding on my generosity once again and that's exactly why I no longer answer anyone on GitHub, clearly this also penalizes users who use SFTPGo without violating its license and/or trademark and I'm very sorry for this.

If you want to offer your customers a file transfer solution, you should write your own and acquire the technical expertise to support it properly, not repackage an open source software.

I'm sorry if the SFTPGo trademark and license are a "problem" for you: they are there to protect my software and my work.

**Comment by snowdream** [comment]
if i package it with License AGPL V3? and keep the project codes with open source. So what is the problem?

**Reply by ralhako** [reply]
The problem is selling freely usable software under your own name. The solution would be simple: a contract with the author and the author's participation in the sale.

**Reply by snowdream** [reply]
Thank you.

**Comment by drakkan** [comment, marked as answer]
For the first time, this was legally enforced with the recent removal of 31 unauthorized listings from the Azure and AWS marketplaces due to intellectual property infringement.

More [details](https://sftpgo.com/intellectual-property-violations).
